metadata-ingestion/src/datahub/ingestion/api/source_helpers.py

```diff
@@ -150,7 +150,7 @@ def auto_workunit_reporter(report: "SourceReport", stream: Iterable[T]) -> Itera
         report.report_workunit(wu)
         yield wu
 
-    if report.events_produced == 0:
+    if report.event_not_produced_warn and report.events_produced == 0:
```

Comment:
I don't fully understand this `event_not_produced_warn` param.

`event_not_produced_warn` looks like a input config param, however it is defined in the `SourceReport` class 🤔 

Also, default value in the `dataclass` is `True` while the field is set `False` in the constructor of `DataHubGcSource`. So this is always `False`, so never raising the warning.

Reply:
Got it, thanks!

May we set `event_not_produced_warn=False` in the initialization of the `DataHubGcSourceReport` itself?

https://github.com/datahub-project/datahub/blob/28cffe3c6a3a3fedce1186ef5fbea7cdda7d0cc6/metadata-ingestion/src/datahub/ingestion/source/gc/datahub_gc.py#L93-L100

Reply:
Done. Thanks for the suggestion